### PR TITLE
refactor: kausal.libsonnet to match k8s-alpha

### PR DESCRIPTION
**Pull request by Duologic**
Essentially a refactoring of kausal.libsonnet to work with k8s-alpha. This might break compat with ksonnet-lib, therefore moved to a new path/file.

ref: #215

(replaces #290)

Some background information: https://gist.github.com/Duologic/875c374cd0df5a04d059fc3e5de9a3dd#gistcomment-3384919

Depends on: https://github.com/jsonnet-libs/k8s/pull/8

### CODE REVIEW

File: k8s-mixin/defaults.libsonnet [hunk 1]
```diff
@@ -33,8 +33,14 @@
       // Expose volumes type.
       volume+:: {
         // Make items parameter optional from fromConfigMap
-        fromConfigMap(name, configMapName, configMapItems={})::
-          super.fromConfigMap(name, configMapName, configMapItems),
+        fromConfigMap(name, configMapName, configMapItems=[])::
+          {
+            configMap+:
+              if configMapItems == [] then { items:: '' }
```
Comment (malcolmholmes) [inline k8s-mixin/defaults.libsonnet:39]: `items:: null` might say more clearly that this will not be a thing

File: k8s-mixin/defaults.libsonnet [hunk 2]
```diff
@@ -233,7 +233,8 @@
         local serviceAccount = $.core.v1.serviceAccount,
 
         service_account:
-          serviceAccount.new(name),
+          serviceAccount.new(name) +
+          serviceAccount.metadata.withNamespace($._config.namespace),
```
Comment (malcolmholmes) [inline k8s-mixin/defaults.libsonnet:248]: is `$._config.namespace` an expectation of this `k8s-mixin`? It isn't an expectation of `k8s-alpha`, is it?

Comment (Duologic) [reply k8s-mixin/defaults.libsonnet:248]: yeah, that would be part of splitting out the utils into a separate util lib that needs its own config

**Comment by Duologic** [comment]
replaced by #426

**Pull request closed without merging**